tools/SourceKit/tools/sourcekitd/lib/API/CodeCompletionResultsArray.cpp

```diff
@@ -56,10 +56,15 @@ void CodeCompletionResultsArrayBuilder::add(
     UIdent SemanticContext,
     UIdent TypeRelation,
     bool NotRecommended,
+    bool IsSystem,
     unsigned NumBytesToErase) {
 
-  assert(NumBytesToErase <= (uint8_t(-1) >> 1));
-  uint8_t BytesAndNotRecommended = (NumBytesToErase << 1) | NotRecommended;
+  assert(NumBytesToErase <= (unsigned(-1) << 2));
+  unsigned BytesAndFlags = 0;
+  BytesAndFlags |= NumBytesToErase << 2;
+  BytesAndFlags |= NotRecommended << 1;
+  BytesAndFlags |= IsSystem << 0;
```

Comment:
I just noticed each value is just casted to pointer size:
https://github.com/apple/swift/blob/eaa1051dfa269a3b6fe7418796140f9f313480c0/tools/SourceKit/tools/sourcekitd/lib/API/CompactArray.cpp#L22-L39
So I decided to use `unsigned` as a bitfield.